tensorflow/java/src/test/java/org/tensorflow/OperationTest.java

```diff
@@ -46,6 +52,96 @@ public void outputListLengthFailsOnInvalidName() {
   }
 
   @Test
+  public void operationEquality() {
+    Operation op1;
+    try (Graph g = new Graph()) {
+      op1 = TestUtil.constant(g, "op1", 1).op();
+      Operation op2 = TestUtil.constant(g, "op2", 2).op();
+      Operation op3 = new Operation(g, op1.getUnsafeNativeHandle());
+      Operation op4 = g.operation("op1");
+      assertEquals(op1, op1);
+      assertNotEquals(op1, op2);
+      assertNotSame(op1, op3);
+      assertEquals(op1, op3);
+      assertEquals(op1.hashCode(), op3.hashCode());
+      assertNotSame(op1, op4);
+      assertNotSame(op3, op4);
+      assertEquals(op1, op4);
+      assertEquals(op1.hashCode(), op4.hashCode());
+      assertEquals(op3, op4);
+      assertNotEquals(op2, op3);
+      assertNotEquals(op2, op4);
+    }
+    try (Graph g = new Graph()) {
+      Operation newOp1 = TestUtil.constant(g, "op1", 1).op();
+      assertNotEquals(op1, newOp1);
+    }
+  }
+
+  @Test
+  public void operationCollection() {
+    try (Graph g = new Graph()) {
+      Operation op1 = TestUtil.constant(g, "op1", 1).op();
+      Operation op2 = TestUtil.constant(g, "op2", 2).op();
+      Operation op3 = new Operation(g, op1.getUnsafeNativeHandle());
+      Operation op4 = g.operation("op1");
+      Set<Operation> ops = new HashSet<>();
+      ops.addAll(Arrays.asList(op1, op2, op3, op4));
+      assertEquals(2, ops.size());
+      assertTrue(ops.contains(op1));
+      assertTrue(ops.contains(op2));
+      assertTrue(ops.contains(op3));
+      assertTrue(ops.contains(op4));
+    }
+  }
+
+  @Test
+  public void operationToString() {
+    try (Graph g = new Graph()) {
+      Operation op = TestUtil.constant(g, "c", new int[] {1}).op();
+      assertEquals("<Const 'c'>", op.toString());
```

Comment:
Do we want to specify a format, or just make sure it doesn't crash?
Since the `toString()` is more of a logging/debugging aid, I'd suggest that we don't test for a specific format and just make sure that the function returns something?